spinalcordtoolbox/moco/dl/inference.py

```diff
@@ -0,0 +1,223 @@
+                continue
+
+            raw_smooth = smooth(img_raw.astype(np.float32), sigmas=[0.5, 0.5])
+            texture = img_raw - raw_smooth
+            out = img_warped + 1.2 * texture
+            lo, hi = np.percentile(img_raw[mask_slice > 0], [0.5, 99.5])
+            out = np.clip(out, lo, hi)
+            sharpened[..., d, t] = out
+
+    matched = np.zeros_like(sharpened)
+    for t in range(T):
+        matched[..., t] = ski_exposure.match_histograms(
+            sharpened[..., t].astype(np.float32),
+            mov_img[..., t].astype(np.float32)
+        )
+
+    # Save Moco output
+    im_moco = Image(matched, hdr=header)
+    im_moco.affine = affine
+    fname_moco_tmp = os.path.join(path_tmp, "mocoDL.nii.gz")
+    im_moco.save(fname_moco_tmp)
+
+    # Save mean output
+    printv("\n[moco-dl] Computing time-averaged output volume...")
+    if mode == "dmri":
+        args = ['-i', fname_moco_tmp, '-bvec', fname_bvecs, '-a', '1', '-v', '0']
+        if fname_bvals:
+            args += ['-bval', fname_bvals]
+        _, fname_b0_mean, _, fname_dwi_mean = sct_dmri_separate_b0_and_dwi.main(argv=args)
+    else:
+        fname_moco_mean = add_suffix(fname_moco_tmp, '_mean')
+        im_moco.mean(dim=3).save(fname_moco_mean)
+
+    # Save Tx, Ty, and displacement fields
+    printv("\n[moco-dl] Saving translation params...")
+    im_Tx = Image(Tx[np.newaxis, np.newaxis, ...], hdr=header)
+    im_Tx.hdr.set_data_shape(im_Tx.data.shape)
+    tx_tmp = os.path.join(path_tmp, "Tx.nii.gz")
+    im_Tx.save(tx_tmp)
+    im_Ty = Image(Ty[np.newaxis, np.newaxis, ...], hdr=header)
+    im_Ty.hdr.set_data_shape(im_Ty.data.shape)
+    ty_tmp = os.path.join(path_tmp, "Ty.nii.gz")
+    im_Ty.save(ty_tmp)
+
+    # Save 5D dispfield
+    printv("\n[moco-dl] Saving displacement fields...")
+    disp5D = np.moveaxis(flow, 0, -1)  # (H,W,D,T,3)
+    im_disp5D = Image(disp5D, hdr=header)
+    im_disp5D.hdr.set_data_shape(disp5D.shape)
+    im_disp5D.hdr.set_intent('vector', (), '')
+    im_disp5D.affine = affine
+    disp_tmp = os.path.join(path_tmp, "displacement-field.nii.gz")
+    im_disp5D.save(disp_tmp)
+
+    # Generate output files
+    printv('\nGenerate output files...')
+    # motion corrected data
+    fname_moco = os.path.join(path_out_abs, add_suffix(os.path.basename(fname_data), "_mocoDL"))
+    generate_output_file(fname_moco_tmp, fname_moco)
+    # mean volume
+    if mode == "dmri":
+        generate_output_file(fname_b0_mean, add_suffix(fname_moco, "_b0_mean"))
+        generate_output_file(fname_dwi_mean, add_suffix(fname_moco, "_dwi_mean"))
+    else:
+        generate_output_file(fname_moco_mean, add_suffix(fname_moco, "_mean"))
+    # rigid translation parameter (Tx, Ty)
+    generate_output_file(tx_tmp, add_suffix(fname_moco, "_Tx"))
+    generate_output_file(ty_tmp, add_suffix(fname_moco, "_Ty"))
+    # 5D displacement field
+    generate_output_file(disp_tmp, add_suffix(fname_moco, "_dispfield"))
```

Comment:
I wonder if we should adopt the same conventions that we use for other warping fields generated by e.g. `sct_register_to_template` and `sct_register_multimodal` -- `warp_{source}2{target}.nii.gz` where `{source}` and `{target}` are the input filenames... Maybe that will be more familiar to users?

So, for example, in the testing data, this would produce a file called `warp_dmri2dwi_mean.nii.gz`. Hm, that's actually a bit confusing looking.

Alternatively, we could just call it `warp_{source}_moco.nii.gz`. So, in this case, `warp_dmri_moco.nii.gz`?